### PR TITLE
feat: ignore sudo error on ACK

### PR DESCRIPTION
**Pull request by albertandrejev**
In this PR tests solution for the issue https://github.com/neutron-org/neutron/pull/84 from the audit report will be added.

More info: https://p2pvalidator.atlassian.net/browse/NTRN-253

### CODE REVIEW

File: src/testcases/interchaintx.test.ts [hunk 1]
```diff
@@ -351,6 +352,78 @@ describe('Neutron / Interchain TXs', () => {
         },
       ]);
     });
+
+    test('delegate with sudo failure', async () => {
+      await cleanAckResults(cm1, contractAddress);
+      // Wait for previous transaction to be executed
+      await waitBlocks(cm1.sdk, 3);
```
Comment (sotnikov-s) [inline src/testcases/interchaintx.test.ts:359]: it doesn't seem to be necessary for there is a waiting for it inside the cleanAckResults (in the underlying execTx if to be precise)

File: src/testcases/simple.test.ts [hunk 2]
```diff
@@ -89,5 +94,61 @@ describe('Neutron / Simple', () => {
         )?.amount,
       ).toEqual('3000');
     });
+
+    test('execute contract with failing sudo', async () => {
+      // Wait for previous transaction to be executed
+      await waitBlocks(cm.sdk, 3);
```
Comment (sotnikov-s) [inline src/testcases/simple.test.ts:100]: same here about the waiting

**Review by oopcode — DISMISSED** [review]

File: src/testcases/interchaintx.test.ts [hunk 3]
```diff
@@ -351,6 +352,78 @@ describe('Neutron / Interchain TXs', () => {
...
+            timeout: 1,
+          },
+        }),
+      );
+
+      const failuresAfterCall = await getWithAttempts<AckFailuresResponse>(
+        cm1.sdk,
+        async () => cm1.queryAckFailures(contractAddress),
+        // Wait until there 1 failure in the list
+        (data) => data.failures.length == 2,
```
Comment (sotnikov-s) [inline src/testcases/interchaintx.test.ts:403]: I'm having a race condition here: sometimes it passes, sometimes it doesn't with the following error:
```
    expected 2, got 1 : [
      {
        address: 'neutron14hj2tavq8fpesdwxxcu44rty3hh90vhujrvcmstl4zr3txmfvw9s5c2epq',
        id: '0',
        ack_id: '3',
        ack_type: 'timeout'
      }
    ]
^^^^^^^^^  THIS IS MY console.log('expected 2, got', data.failures.length, ':', data.failures);

    getWithAttempts: no attempts left

      51 |     await waitBlocks(sdk, 1);
      52 |   }
    > 53 |   throw error != null ? error : new Error('getWithAttempts: no attempts left');
         |                                 ^
      54 | };
      55 |

      at getWithAttempts (src/helpers/wait.ts:53:33)
          at runMicrotasks (<anonymous>)
      at Object.<anonymous> (src/testcases/interchaintx.test.ts:399:33)
```
i.e. it seems like sometimes the sudo failure mock doesn't work. this is true for both cases with the waiting I concerned about and without it

**Review by sotnikov-s — CHANGES_REQUESTED** [review]

File: src/testcases/interchaintx.test.ts (hunk 3 above)
Comment (sotnikov-s) [reply src/testcases/interchaintx.test.ts:403]: it's also sometimes like this: there is a failure for ack, but no failure for timeout
```
    expected 2, got 1 : [
      {
        address: 'neutron14hj2tavq8fpesdwxxcu44rty3hh90vhujrvcmstl4zr3txmfvw9s5c2epq',
        id: '0',
        ack_id: '2',
        ack_type: 'ack'
      }
    ]
```

**Review by pr0n00gler — DISMISSED** [review]

**Review by oopcode — APPROVED** [review]
tests pass

**Review by sotnikov-s — APPROVED** [review]

**Pull request merged**